src/Types/ConnectionType.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+namespace Nuwave\Relay\Types;
+
+use GraphQL;
+use Closure;
+use Folklore\GraphQL\Support\Type as GraphQLType;
+use GraphQL\Type\Definition\ObjectType;
+use GraphQL\Type\Definition\ResolveInfo;
+use GraphQL\Type\Definition\Type;
+use Illuminate\Contracts\Pagination\LengthAwarePaginator;
+use Nuwave\Relay\Traits\GlobalIdTrait;
+
+abstract class ConnectionType extends GraphQLType
+{
+
+    use GlobalIdTrait;
+
+    /**
+     * The edge resolver for this connection type
+     *
+     * @var \Closure
+     */
+    protected $edgeResolver;
+
+    /**
+     * The pageInfo resolver for this connection type.
+     *
+     * @var \Closure
+     */
+    protected  $pageInfoResolver;
+
+
+    /**
+     * The name of the edge (i.e. `User`).
+     *
+     * @var string
+     */
+    protected $name = '';
+
+    /**
+     * Special fields present on this connection type.
+     *
+     * @return array
+     */
+    public function fields()
+    {
+        return [];
+    }
+
+    /**
+     * Fields that exist on every connection.
+     *
+     * @return array
+     */
+    protected function baseFields()
+    {
+        return [
+            'pageInfo' => [
+                'type' => Type::nonNull(GraphQL::type('pageInfo')),
+                'description' => 'Information to aid in pagination.',
+                'resolve' => function ($collection) {
+                    return $collection;
+                },
+            ],
+            'edges' => [
+                'type' => Type::listOf($this->buildEdgeType($this->name, $this->type())),
+                'description' => 'Information to aid in pagination.',
+                'resolve' => function ($collection) {
+                    return $this->injectCursor($collection);
+                },
+            ]
+        ];
+    }
+
+    /**
+     * Get the default arguments for a connection.
+     *
+     * @return array
+     */
+    public static function connectionArgs()
+    {
+        return [
+            'after' => [
+                'type' => Type::string()
+            ],
+            'first' => [
+                'type' => Type::int()
+            ],
+            'before' => [
+                'type' => Type::string()
+            ],
+            'last' => [
+                'type' => Type::int()
+            ]
+        ];
+    }
+
+    /**
+     * Build the edge type for this connection.
+     *
+     * @param $name
+     * @param $type
+     * @return ObjectType
+     */
+    protected function buildEdgeType($name, $type)
+    {
```

Comment:
The [naming convention](https://facebook.github.io/relay/graphql/connections.htm) of Relay edges would be "ExampleName" rather than "ExampleConnectionName", so we could do something like:

```php
if (preg_match('/Connection$/', $name)) {
    $name = substr($name, 0, strlen($name) - 10);
}
```